src/languages/asciidoc.js

```diff
@@ -107,17 +138,55 @@ export default function(hljs) {
         begin: '^(NOTE|TIP|IMPORTANT|WARNING|CAUTION):\\s+',
         relevance: 10
       },
-      // inline strong
+      // inline unconstrained strong (single line)
       {
         className: 'strong',
-        // must not follow a word character or be followed by an asterisk or space
-        begin: '\\B\\*(?![\\*\\s])',
-        end: '(\\n{2}|\\*)',
-        // allow escaped asterisk followed by word char
-        contains: [{
-          begin: '\\\\*\\w',
-          relevance: 0
-        }]
+        begin: `\\*{2}(${CC_ANY}+?)\\*{2}`
+      },
+      // inline unconstrained strong (multi-line)
+      {
+        className: 'strong',
+        begin: /\*{2}(([^*\n\\]|\\[^\n])+\n)+([^*\n\\]|\\[^\n])*\*{2}/,
+        relevance: 0
+      },
+      // escaped constrained formatting marks (i.e., \* \_ or \`)
+      {
+        begin: /\\[*_`]/
+      },
+      // escaped unconstrained formatting marks (i.e., \\** \\__ or \\``)
+      // must ignore until the next formatting marks
+      // this rule is not 100% compliant with Asciidoctor but we are entering undefined behavior territory...
+      {
+        begin: /\\\\\*{2}[^\n]*\*{2}/
+      },
+      {
+        begin: /\\\\_{2}[^\n]*_{2}/
+      },
+      {
+        begin: /\\\\`{2}[^\n]*`{2}/
+      },
+      // constrained preceded by ":", ";", or "}".
+      {
+        begin: /[:;}][*_`](?!\*)/
+      },
+      // inline constrained string (single line)
+      {
+        className: 'strong',
+        // must not precede or follow a word character
+        begin: `\\B\\*(\\S|\\S${CC_ANY}*?\\S)\\*(?!${CG_WORD})`,
+      },
+      // inline constrained strong (multi-line)
+      {
+        className: 'strong',
+        // must not precede or follow a word character
+        begin: /\B\*(\w\n?)*\*(?!\w)/,
```

Comment:
I'm so confused about constrains vs unconstrained... should this work:

```
*does bob
-really-
know best?*
```

Reply:
> I'm so confused about constrains vs unconstrained... shouldn't this work:

```
*does bob
-really-
know best?*
```

The above should produce bold.

> I'm wondering if perhaps this is really just the same as the ** except for the before/end conditions? If so then that regex would probably work here also I think?

Close but not quite the same: https://asciidoctor.org/docs/user-manual/#when-should-i-use-unconstrained-quotes

> Consider the following questions:
> Is there a letter, number, underscore directly outside the formatting marks (on either side)?
> Is there a colon, semi-colon, or closing curly bracket directly before the starting formatting mark?
> Is there a space directly inside of the formatting mark?
> If you answered “yes” to any of these questions, you need to switch to unconstrained (double formatting) quotes.

See: https://github.com/asciidoctor/asciidoctor/blob/dd843d59b1e49f29d68386bfdf0438ce11b42577/lib/asciidoctor.rb#L447-L449
Please keep in mind that we are also doing some work in https://github.com/asciidoctor/asciidoctor/blob/dd843d59b1e49f29d68386bfdf0438ce11b42577/lib/asciidoctor/substitutors.rb#L75

Reply:
I'm also using a less greedy regular expression than Asciidoctor because we want to stop when the paragraph ends.

Reply:
> The above should produce bold.

But your rule doesn't match the `-`, or am I missing something?

> Close but not quite the same

Unless I'm missing something all I read there relates directly to what qualifies the beginning and ending of the match though, yes?  Matching the "middle" content could use the same (or similar) wider rule so that it wouldn't get hung up on details like `-` or punctuation, etc.  So really we're trying to structure the regex into 3 segments:

- beginning
- middle
- end

It seems for both `*` and `**` the middle rules are the same, no?  Only the begin and end conditions change?

Reply:
> But your rule doesn't match the -, or am I missing something?

Oh yes, you're right, sorry I didn't understand.
I've added this case and updated the regular expression accordingly.

> Unless I'm missing something all I read there relates directly to what qualifies the beginning and ending of the match though, yes?

No, this rule is about what is inside the formatting marks: "Is there a space directly inside of the formatting mark?"

```
this is * a bold move*

this is ** a bold move**
```

Only the second sentence will produce bold.

> So really we're trying to structure the regex into 3 segments:

I guess since the formatting marks are delimiters, so:

- beginning: delimiter
- middle: some content
- end: delimiter

In theory, that should be easy but in practice it's not because "*" can be used in different *context*:

```
****
sidebar
****

* foo
* bar
** baz

Read the footnotes*.

E = mc*2*

E = *mc*2

Use `/*` and `*/` for multiline comments.
```

The last example is not correctly interpreted by Asciidoctor :sweat_smile: 

Anyway, we should be careful about "as WIDE as possible".